src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -800,20 +800,54 @@ private void ConfigureLogging()
 
             _hostConfig.Tracing.Tracers.Add(_traceMonitor);
             var hostTraceLevel = _hostConfig.Tracing.ConsoleLevel;
+            IList<TraceWriter> traceWriters = new List<TraceWriter>();
             if (ScriptConfig.FileLoggingMode != FileLoggingMode.Never)
             {
                 // Host file logging is only done conditionally
                 string hostLogFilePath = Path.Combine(ScriptConfig.RootLogPath, "Host");
                 TraceWriter fileTraceWriter = new FileTraceWriter(hostLogFilePath, hostTraceLevel, LogType.Host).Conditional(p => FileLoggingEnabled);
+                traceWriters.Add(fileTraceWriter);
+            }
+
+            // See if there is any registered TraceWriterFactory. This will create an additional TraceWriter to add to this host.
+            IHostTraceWriterFactory hostTraceWriterFactory = ScriptConfig.HostConfig.GetService<IHostTraceWriterFactory>();
+            if (hostTraceWriterFactory != null)
+            {
+                // Pre-parse the host.json to apply the ConsoleLevel value here. Default to Info if nothing is found.
+                // Any parse errors are ignored and will be caught and logged later.
+                TraceLevel level = TraceLevel.Info;
+                try
+                {
+                    string json = File.ReadAllText(_hostConfigFilePath);
+                    JObject hostConfigObject = JObject.Parse(json);
 
+                    var traceLevelStr = hostConfigObject["tracing"]?["consoleLevel"]?.ToString();
+                    if (Enum.TryParse(traceLevelStr, true, out TraceLevel parsedLevel))
+                    {
+                        level = parsedLevel;
+                    }
+                }
+                catch
+                {
+                }
+
+                TraceWriter hostTraceWriter = hostTraceWriterFactory.Create(level);
+
+                if (hostTraceWriter != null)
+                {
+                    traceWriters.Add(hostTraceWriter);
+                }
+            }
```

Comment:
@fabiocav -- thoughts on this `IHostTraceWriterFactory` approach? 

I wanted:
1. 'consoleLevel' to apply to this TraceWriter. That means we need to read the host.json before creating TraceWriters. Without a major refactor...
2. To keep all of our diagnostic/ETW stuff in WebHost. I could have moved a lot of that here and gotten rid of that factory, but figured this may be less disruptive.
3. Not get *too* disruptive with the factory -- so I didn't create a `DefaultHostTraceWriterFactory` and move the `FileTraceWriter` into it, for example.